bin/reth/src/lib.rs

```diff
@@ -43,3 +46,79 @@ struct NetworkOpts {
     #[arg(long, value_delimiter = ',')]
     bootnodes: Option<Vec<NodeRecord>>,
 }
+
+/// Parameters for configuring the network more granularly via CLI
+#[derive(Debug, Args, PartialEq, Default)]
+#[command(next_help_heading = "Rpc")]
+struct RpcServerOpts {
+    /// Enable the HTTP-RPC server
+    #[arg(long)]
+    http: bool,
+
+    /// Http server address to listen on
+    #[arg(long = "http.addr")]
+    http_addr: Option<IpAddr>,
+
+    /// Http server port to listen on
+    #[arg(long = "http.port")]
+    http_port: Option<u16>,
+
+    /// Rpc Modules to be configured for http server
+    #[arg(long = "http.api")]
+    http_api: Option<RpcModuleConfig>,
+
+    /// Enable the WS-RPC server
+    #[arg(long)]
+    ws: bool,
+
+    /// Ws server address to listen on
+    #[arg(long = "ws.addr")]
+    ws_addr: Option<IpAddr>,
+
+    /// Http server port to listen on
+    #[arg(long = "ws.port")]
+    ws_port: Option<u16>,
+
+    /// Rpc Modules to be configured for Ws server
+    #[arg(long = "ws.api")]
+    ws_api: Option<RpcModuleConfig>,
+
+    /// Disable the IPC-RPC  server
+    #[arg(long)]
+    ipcdisable: bool,
+
+    /// Filename for IPC socket/pipe within the datadir
+    #[arg(long)]
+    ipcpath: Option<String>,
+}
+
+impl RpcServerOpts {
+    fn parse_from(input: Vec<&str>) -> Option<RpcServerOpts> {
+        let cmd = Command::new("reth").arg(
+            Arg::new("http_api").long("http.api").value_parser(value_parser!(RpcModuleConfig)),
+        );
+
+        let mut server_opts = RpcServerOpts::default();
+
+        if let Some(api) = cmd.get_matches_from(input).get_one::<RpcModuleConfig>("http_api") {
+            server_opts.http_api = Some(api.clone());
+        };
+
+        Some(server_opts)
+    }
+}
```

Comment:
you don't need this. 
This is provided by the `Parser` derive macro https://docs.rs/clap/latest/clap/trait.Parser.html